drivers/display/display_nt35510.c

```diff
@@ -0,0 +1,417 @@
+
+struct nt35510_config {
+	const struct device *mipi_dsi;
+	const struct gpio_dt_spec reset;
+	const struct gpio_dt_spec backlight;
+	uint8_t data_lanes;
+	uint16_t width;
+	uint16_t height;
+	uint8_t channel;
+	uint16_t rotation;
+};
+
+struct nt35510_data {
+	enum display_pixel_format dsi_pixel_format;
+	enum display_orientation orientation;
+	uint16_t xres;
+	uint16_t yres;
+};
+
+/* Write data buffer to LCD register */
+static int nt35510_write_reg(const struct device *dev, uint8_t reg, const uint8_t *buf, size_t len)
+{
+	int ret;
+	const struct nt35510_config *cfg = dev->config;
+
+	ret = mipi_dsi_dcs_write(cfg->mipi_dsi, cfg->channel, reg, buf, len);
+	if (ret < 0) {
+		LOG_ERR("Failed writing reg: 0x%x result: (%d)", reg, ret);
+		return ret;
+	}
+	return 0;
+}
+
+/* Write an 8-bit value to a register */
+static int nt35510_write_reg_val(const struct device *dev, uint8_t reg, uint8_t value)
+{
+	return nt35510_write_reg(dev, reg, &value, 1);
+}
+
+/* Initialization and configuration sequence */
+static int nt35510_config(const struct device *dev)
+{
+	struct nt35510_data *data = dev->data;
+	int ret = 0;
+
+	static const uint8_t nt35510_reg1[] = {0x03, 0x03, 0x03};
+	static const uint8_t nt35510_reg2[] = {0x46, 0x46, 0x46};
+	static const uint8_t nt35510_reg[] = {0x55, 0xAA, 0x52, 0x08, 0x01};
+	static const uint8_t nt35510_reg3[] = {0x03, 0x03, 0x03};
+	static const uint8_t nt35510_reg4[] = {0x36, 0x36, 0x36};
+	static const uint8_t nt35510_reg5[] = {0x00, 0x00, 0x02};
+	static const uint8_t nt35510_reg6[] = {0x26, 0x26, 0x26};
+	static const uint8_t nt35510_reg7[] = {0x01, 0x01};
+	static const uint8_t nt35510_reg8[] = {0x09, 0x09, 0x09};
+	static const uint8_t nt35510_reg9[] = {0x36, 0x36, 0x36};
+	static const uint8_t nt35510_reg10[] = {0x08, 0x08, 0x08};
+	static const uint8_t nt35510_reg12[] = {0x26, 0x26, 0x26};
+	static const uint8_t nt35510_reg13[] = {0x00, 0x80, 0x00};
+	static const uint8_t nt35510_reg14[] = {0x00, 0x80, 0x00};
+	static const uint8_t nt35510_reg15[] = {0x00, 0x50};
+	static const uint8_t nt35510_reg16[] = {0x55, 0xAA, 0x52, 0x08, 0x00};
+	static const uint8_t nt35510_reg17[] = {0xFC, 0x00};
+	static const uint8_t nt35510_reg18[] = {0x03, 0x03};
+	static const uint8_t nt35510_reg19[] = {0x51, 0x51};
+	static const uint8_t nt35510_reg20[] = {0x00, 0x00};
+
+	static const uint8_t nt35510_reg21[] = {0x01, 0x02, 0x02, 0x02};
+	static const uint8_t nt35510_reg22[] = {0x00, 0x00, 0x00};
+	static const uint8_t nt35510_reg23[] = {0x03, 0x00, 0x00};
+	static const uint8_t nt35510_reg24[] = {0x01, 0x01};
+
+	static const uint8_t nt35510_caset_portrait[] = {0x00, 0x00, 0x01, 0xDF};
+	static const uint8_t nt35510_raset_portrait[] = {0x00, 0x00, 0x03, 0x1F};
+	static const uint8_t nt35510_caset_landscape[] = {0x00, 0x00, 0x03, 0x1F};
+	static const uint8_t nt35510_raset_landscape[] = {0x00, 0x00, 0x01, 0xDF};
+
+	ret = nt35510_write_reg(dev, 0xF0, nt35510_reg, 5);   /* LV2:  Page 1 enable */
```

Comment:
I'd prefer to avoid using this method of checking the return code- it makes it difficult for a developer to see where the write actually failed when debugging (since they would need to check the value of `ret` each time).

Maybe put these initialization commands into a const array, and loop over that array? Something like the following:

```
struct nt35510_init_cmd {
    uint8_t reg;
    uint8_t cmd_len;
    uint8_t cmd[]
};

static const struct nt35510_init_cmd init_cmds[] = {
{
    .reg = 0xF0,  /* LV2:  Page 1 enable */
    .cmd_len = 5,
    .cmd = {0x03, 0x03, 0x03},
};

....

struct nt35510_init_cmd *cmd;
for (int i = 0; i < ARRAY_SIZE(init_cmds); i++) {
    cmd = init_cmds[i];
    ret = nt35510_write_reg(dev, cmd->reg, cmd->cmd, cmd->cmd_len);
    if (ret < 0) {
        return ret;
    }
}
```

Reply:
Agree, will possibly occupy less program space too
I will borrow your example above and extend it with a delay struct member needed for those two
delays required in initialization sequence